### PR TITLE
when line is too long, bufio.ReadLine may not read all data

### PR DESCRIPTION
**Pull request by CodeRushing**
when line is too long, bufio.ReadLine may not read all data

### CODE REVIEW

File: ogórek.go [hunk 1]
```diff
@@ -238,6 +238,22 @@ func (d Decoder) Decode() (interface{}, error) {
 	return d.pop(), nil
 }
 
+func (d *Decoder) readLine() ([]byte, error) {
+	var has_more bool = true
```
Comment (kisielk) [inline ogórek.go]: just call this isPrefix and assign to it in L245. You can predeclare `readData` and `err` as well. Also please use camelCase.

File: ogórek.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
-// Package ogórek is a library for decoding Python's pickle format.
+// Package gopickle is a library for decoding Python's pickle format.
 //
-// ogórek is Polish for "pickle".
-package ogórek
+// gopickle is Polish for "pickle".
+package gopickle
```
Comment (kisielk) [inline ogórek.go]: please don't change the package name.

**Comment by kisielk** [comment]
A test case for this would also be good.

**Comment by CodeRushing** [comment]
I have add a test case for this and I'm sorry about the package name

File: ogórek.go [hunk 3]
```diff
@@ -238,6 +238,22 @@ func (d Decoder) Decode() (interface{}, error) {
...
+	var has_more bool = true
+	var line []byte = []byte{}
+	for has_more {
+		read_data, is_prefix, err := d.r.ReadLine()
+		if err != nil {
+			return line, err
+		}
+		has_more = is_prefix
+		for i := 0; i < len(read_data); i++ {
+			line = append(line, read_data[i])
```
Comment (kisielk) [inline ogórek.go]: as Damian mentioned, 

`line = append(line, readData...)` and you can remove the loop.

File: ogórek.go [hunk 4]
```diff
@@ -1,6 +1,6 @@
-// Package ogórek is a library for decoding Python's pickle format.
+// Package gopickle is a library for decoding Python's pickle format.
```
Comment (kisielk) [inline ogórek.go]: please undo the comment changes as well.

Comment (CodeRushing) [inline ogórek.go]: ok, I've finished that

**Pull request merged**